scripts/missions/asa/04_Sugar-Coated_Directive.lua

```diff
@@ -0,0 +1,277 @@
+
+                onEventFinish =
+                {
+                    [2] = function(player, csid, option, npc)
+                        player:delKeyItem(xi.ki.DOMINAS_AMBER_SEAL)
+                        player:addKeyItem(xi.ki.AMBER_COUNTERSEAL)
+                        player:messageSpecial(tremorsID.text.ATTACH_SEAL, xi.ki.DOMINAS_AMBER_SEAL)
+                        player:messageSpecial(tremorsID.text.KEYITEM_OBTAINED, xi.ki.AMBER_COUNTERSEAL)
+                        local sealCount = mission:getVar(player, 'SealCount')
+                        mission:setVar(player, 'SealCount', sealCount + 1)
+                    end,
+                }
+            },
+        },
+
+        [xi.zone.QUFIM_ISLAND] =
+        {
+            ['Trodden_Snow'] =
+            {
+                onTrigger = function(player, npc)
+                    --ASA 4 CS: Triggers With At Least 3 Counterseals.
+                    local completedSeals =
+                        (player:hasKeyItem(xi.ki.AMBER_COUNTERSEAL)    and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.AZURE_COUNTERSEAL)    and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.CERULEAN_COUNTERSEAL) and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.EMERALD_COUNTERSEAL)  and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.SCARLET_COUNTERSEAL)  and 1 or 0) +
+                        (player:hasKeyItem(xi.ki.VIOLET_COUNTERSEAL)   and 1 or 0)
+
+                    if completedSeals >= 3 then
+                        mission:setVar(player, 'SealCount', completedSeals)
+                        return mission:progressEvent(45)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [45] = function(player, csid, option, npc)
+                    local completedSeals = mission:getVar(player, 'SealCount')
+
+                    -- Calculate Reward
+                    if completedSeals == 3 then
+                        player:addGil(xi.settings.GIL_RATE * 3000)
+                        player:messageSpecial(qufimID.text.GIL_OBTAINED, xi.settings.GIL_RATE * 3000)
+                    elseif completedSeals == 4 then
+                        player:addGil(xi.settings.GIL_RATE * 10000)
+                        player:messageSpecial(qufimID.text.GIL_OBTAINED, xi.settings.GIL_RATE * 10000)
+                    elseif completedSeals == 5 then
+                        player:addGil(xi.settings.GIL_RATE * 30000)
+                        player:messageSpecial(qufimID.text.GIL_OBTAINED, xi.settings.GIL_RATE * 30000)
+                    elseif completedSeals == 6 then
+                        player:addGil(xi.settings.GIL_RATE * 50000)
+                        player:messageSpecial(qufimID.text.GIL_OBTAINED, xi.settings.GIL_RATE * 50000)
+                    end
+
+                    -- Clean Up Remaining Key Items
+                    player:delKeyItem(xi.ki.DOMINAS_SCARLET_SEAL)
```

Comment:
Since mentioning tabled KIs previously, this could also be used to clean up these keyitems

Reply:
I think I've got this updated to what you're referring to but please let me know if I got it wrong.